limitador/src/storage/redis/counters_cache.rs

```diff
@@ -189,8 +193,13 @@ impl Batcher {
                 }
                 let result = consumer(result).await;
                 batch.iter().for_each(|counter| {
-                    self.updates
-                        .remove_if(counter, |_, v| v.no_pending_writes());
+                    if self
+                        .updates
+                        .remove_if(counter, |_, v| v.no_pending_writes())
+                        .is_some()
+                    {
+                        self.limiter.add_permits(1);
```

Comment:
I think the example in the doc is somewhat oversimplified...
Reading the docs, `.add_permits` doesn't account for the initial permits provided (i.e. consider it a max), see [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=42c8b23185148572e6111882673eb2c4) 
So we need to do all the math ourselves, as this wouldn't account for re-adding permits back when counters expire for instance. I'd expect this would run out of permit after a while.

Reply:
Actually... thinking about it, it might be good enough. As we don't compact the queue anywhere 🤔 
So that, if the entry is passed to the `consumer`, even if there are no pending writes for this entry e.g. because it expired, then this would also remove it from `self.updates` and increment the tickets back...